torch/_functorch/_aot_autograd/functional_utils.py

```diff
@@ -191,7 +191,49 @@ def has_metadata_mutation(f_arg, arg, *, check_only_storage_mutation: bool):
         return has_metadata_mutation_
 
 
-def gen_alias_from_base(aliased_base_tensor, target_meta_tensor, target_requires_grad):
+def gen_alias_from_base(
+    aliased_base_tensor,
+    target_meta_tensor,
+    target_requires_grad,
+    target_functional_tensor=None,
+):
+    # Patch the correct requires_grad field of the output tensor, depending on whether:
+    # (i) the reconstructed output (out) was came from a tensor that requires grad or not;
+    # and (ii) the concrete returned output does require grad or not.
+    def patch_requires_grad(out):
+        if aliased_base_tensor.requires_grad and not target_requires_grad:
+            out = out.detach()
+        elif not aliased_base_tensor.requires_grad and target_requires_grad:
+            out.requires_grad_(True)
+        return out
+
+    # If provided, use the target functional tensor for replaying the views.
+    #
+    # In summary, we use the fact that FunctionalTensorWrapper saves the view
+    # functions applied to itself (collected during functionalization) so as
+    # to replay them (view functions) on the aliased_base_tensor.
+    if target_functional_tensor is not None:
+        functional_tensor = target_functional_tensor.tensor
+
+        try:
+            out = torch._functionalize_apply_view_metas(
+                functional_tensor, aliased_base_tensor
+            )
+            assert out.shape == target_meta_tensor.shape, (
+                "incorrect out shape after application of ViewMeta sequence: "
+                f"{tuple(out.shape)} (actual) vs {tuple(target_meta_tensor.shape)} (expected)"
+            )
+            return patch_requires_grad(out)
+        except RuntimeError:
```

Comment:
hmm I'm a bit worried about blindly swallowing the error. Although I guess it's fine since we know we can silently fall back to as_strided (the old behavior).

It might make diagnosing `as_strided` issues a pain though. Can you log the exception string? We don't actually have a great dedicated logger for general AOTAutograd info. But it seems... reasonable enough if you use the existing aot joint logger: https://github.com/pytorch/pytorch/blob/main/torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py#L65